src/account/0.7/PasskeyAccount.sol

```diff
@@ -0,0 +1,274 @@
+        returns (bool)
+    {
+        return _passkey.pubKeyX != 0 && _passkey.pubKeyY != 0;
+    }
+
+    /**
+     * @param userOp typical PackedUserOperation
+     * @param userOpHash the hash of the user operation.
+     * @return validationData
+     */
+    function _validateSignature(
+        PackedUserOperation calldata userOp,
+        bytes32 userOpHash // As Webauthn's challenge
+    ) internal view virtual override returns (uint256) {
+        require(_isPasskeyValid(passkey), "Passkey doesn't exist");
+        (
+            bool isSimulation,
+            bytes memory authenticatorData,
+            bool requireUserVerification,
+            string memory clientDataJSON,
+            uint256 challengeLocation,
+            uint256 responseTypeLocation,
+            uint256 r,
+            uint256 s
+        ) = abi.decode(
+            userOp.signature,
+            (bool, bytes, bool, string, uint256, uint256, uint256, uint256)
+        );
+        bytes memory challenge =
+            isSimulation ? bytes("") : abi.encodePacked(userOpHash);
+        bool isSigValid = _verifySignatureWebauthn(
+            challenge,
+            authenticatorData,
+            requireUserVerification,
+            clientDataJSON,
+            challengeLocation,
+            responseTypeLocation,
+            r,
+            s,
+            passkey.pubKeyX,
+            passkey.pubKeyY
+        );
+        if (isSimulation) {
+            return SIG_VALIDATION_FAILED;
+        }
+        return isSigValid ? 0 : SIG_VALIDATION_FAILED;
+    }
+
+    function _verifySignatureAllowMalleability(
+        bytes32 messageHash,
+        uint256 r,
+        uint256 s,
+        uint256 x,
+        uint256 y
+    ) internal view returns (bool) {
+        bytes memory args = abi.encode(messageHash, r, s, x, y);
+        (bool success, bytes memory ret) =
+            address(p256Verifier).staticcall(args);
+        assert(success); // never reverts, always returns 0 or 1
+
+        return abi.decode(ret, (uint256)) == 1;
+    }
+
+    /// P256 curve order n/2 for malleability check
+    uint256 constant P256_N_DIV_2 =
+        57896044605178124381348723474703786764998477612067880171211129530534256022184;
+
+    function _verifySignature(
+        bytes32 messageHash,
+        uint256 r,
+        uint256 s,
+        uint256 x,
+        uint256 y
+    ) internal view returns (bool) {
+        // check for signature malleability
+        if (s > P256_N_DIV_2) {
+            return false;
+        }
+
+        return _verifySignatureAllowMalleability(messageHash, r, s, x, y);
+    }
+
+    /**
+     * Verifies a Webauthn P256 signature (Authentication Assertion) as described
+     * in https://www.w3.org/TR/webauthn-2/#sctn-verifying-assertion. We do not
+     * verify all the steps as described in the specification, only ones relevant
+     * to our context. Please carefully read through this list before usage.
+     */
+    function _verifySignatureWebauthn(
```

Comment:
Extract a `WebAuthnSignatureVerifier` library to `util` and remove the duplication between 0.6 and 0.7